File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if (argc != 2 || argv[1] < 0)
    {
        printf("Error!\n");
        
        return 1;
    }
    
    int k = atoi(argv[1]);

    int length = 0;

    int i = 0;

    printf("plaintext: ");

    string plaint = get_string();

    length = strlen(plaint);

    printf("ciphertext: ");

    for (i = 0; i < length; i++)
    {
        if (isupper(plaint[i]))
        {
            printf("%c", ((((plaint[i] + k) - 65) % 26) + 65));
        }
        else if (islower(plaint[i]))
        {
            printf("%c", ((((plaint[i] + k) - 97) % 26) + 97));
        }
        else
        {
            printf("%c", plaint[i]);
        }
    }
    printf("\n");
    return 0;
}

File: pset6/sentimental/caesar/caesar.py
import cs50
import sys
from cs50 import get_string


def main():
    count = len(sys.argv)

    # Check correct no. of arguments
    if count != 2:
        print("Usage: ./caesar k")
        sys.exit(1)
    else:
        print("plaintext: ", end="")
        instring = get_string()
        print("ciphertext: ", end="")

    # Set input to an int, to be the cipher.
    k = int(sys.argv[1])

    # Iterate through the input string, converting to relevant ASCII values dpending on case. Do not change if not alphabetic.
    for i in instring:
        if i.islower():
            j = ((ord(i) + k - 97) % 26) + 97
            print((chr(j)), end="")
        elif i.isupper():
            j = ((ord(i) + k - 65) % 26) + 65
            print((chr(j)), end="")
        else:
            print(i, end="")

    print()


if __name__ == "__main__":
    main()